src/charm.py

```diff
@@ -714,22 +750,39 @@ def _is_bessd_grpc_service_ready(self) -> bool:
     def _is_bessd_configured(self) -> bool:
         """Check if bessd has been configured.
 
-        Examine the output from bessctl to show worker. If there is no
-        active worker, bessd is assumed not to be configured.
+        Bess is considered as configured when the `accessRoutes` and `coreRoutes` modules
+        are available and at least one worker is in the RUNNING state. This method examines
+        the output of the `bessctl show` command to determine whether the above criteria are met.
 
         Returns:
-            bool:   True/False
+            bool: Whether the `bessd` service is configured
         """
-        command = "/opt/bess/bessctl/bessctl show worker"
+        show_accessRoutes_module_cmd = "/opt/bess/bessctl/bessctl show module accessRoutes"  # noqa: N806
+        show_coreRoutes_module_cmd = "/opt/bess/bessctl/bessctl show module coreRoutes"  # noqa: N806
+        show_worker_cmd = "/opt/bess/bessctl/bessctl show worker"
         try:
-            (stdout, _) = self._exec_command_in_bessd_workload(
-                command=command,
+            (show_accessRoutes_module_stdout, _) = self._exec_command_in_bessd_workload(  # noqa: N806
+                command=show_accessRoutes_module_cmd,
+                timeout=10,
+            )
+            logger.debug(
+                "bessd configured accessRoutes module: %s", show_accessRoutes_module_stdout
+            )
+            (show_coreRoutes_module_stdout, _) = self._exec_command_in_bessd_workload(  # noqa: N806
+                command=show_coreRoutes_module_cmd,
+                timeout=10,
+            )
+            logger.debug("bessd configured coreRoutes module: %s", show_coreRoutes_module_stdout)
+            (show_worker_stdout, _) = self._exec_command_in_bessd_workload(
+                command=show_worker_cmd,
                 timeout=10,
             )
-            logger.debug("bessd configured workers: %s", stdout)
+            if "RUNNING" not in show_worker_stdout:
+                return False
+            logger.debug("bessd configured workers: %s", show_worker_stdout)
             return True
         except ExecError as e:
-            logger.info("Configuration check: %s", e)
+            logger.error("Configuration check failed: %s", e)
```

Comment:
Also the reason why this was not written as error before, is that it is normal for the first check to fail. If the up4 script has not run, this is our indication that it needs to run.

Reply:
I'd still have it as `error`. The problem I have with the `info` logs in such cases is that it's easy to miss them. Errors are printed in red, so it's super easy to find them.